Rubberduck.Deployment.Build/RubberduckPostBuildTask.cs

```diff
@@ -0,0 +1,365 @@
+
+            this.LogMessage("Creating InnoSetup registry entries...");
+            CreateInnoSetupRegistryEntries(entries, parameters);
+
+            if (Config != "Debug")
+            {
+                return;
+            }
+
+            RemovePreviousDebugRegistration();
+            UpdateDebugRegistration(entries, parameters);
+        }
+
+        private void CreateIdlFile(DllFileParameters parameters)
+        {
+            var generator = new IdlGenerator();
+            var idl = generator.GenerateIdl(parameters.SourceDll);
+            // Encoding must be UTF8 BOM
+            File.WriteAllText(Path.Combine(TargetDir, parameters.IdlFile), idl, new UTF8Encoding(true));
+        }
+
+        private void CompileWithMidl(DllFileParameters parameters)
+        {
+            var targetPath = TargetDir.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+            var command = $"call \"{_batchPath}\"{Environment.NewLine}" +
+                          $"midl.exe /win32 /tlb \"{parameters.Tlb32File}\" \"{parameters.IdlFile}\" /out \"{targetPath}\"{Environment.NewLine}" +
+                          $"midl.exe /amd64 /tlb \"{parameters.Tlb32File}\" \"{parameters.IdlFile}\" /out \"{targetPath}\"";
+            ExecuteTask(command, SourceDir, true);
+        }
+
+        private void CompileWithTlbExp(DllFileParameters parameters)
+        {
+            var command = $"{NetToolsDir}tlbexp.exe \"{parameters.SourceDll}\" /win32 /out:\"{parameters.SourceTlb32}\"";
+            ExecuteTask(command);
+
+            command = $"{NetToolsDir}tlbexp.exe \"{parameters.SourceDll}\" /win64 /out:\"{parameters.SourceTlb64}\"";
+            ExecuteTask(command);
+        }
+
+        private void HarvestMetadataWithWix(DllFileParameters parameters)
+        {
+            var command = $"{WixToolsDir}heat.exe file \"{parameters.SourceDll}\" -out \"{parameters.DllXml}\"";
+            ExecuteTask(command);
+
+            command = $"{WixToolsDir}heat.exe file \"{parameters.SourceTlb32}\" -out \"{parameters.TlbXml}\"";
+            ExecuteTask(command);
+        }
+
+        private IOrderedEnumerable<RegistryEntry> BuildRegistryEntriesFromMetadata(DllFileParameters parameters)
+        {
+            var builder = new RegistryEntryBuilder();
+            return builder.Parse(parameters.TlbXml, parameters.DllXml);
+        }
+
+        private void CreateInnoSetupRegistryEntries(IOrderedEnumerable<RegistryEntry> entries, DllFileParameters parameters)
+        {
+            var writer = new InnoSetupRegistryWriter();
+            var content = writer.Write(entries, parameters.DllFile, parameters.Tlb32File, parameters.Tlb64File);
+            var regFile = Path.Combine(IncludeDir, parameters.DllFile.Replace(".dll", ".reg.iss"));
+            
+            // Encoding must be UTF8 BOM
+            File.WriteAllText(regFile, content, new UTF8Encoding(true));
+        }
+
+        private bool _previousRegistrationRemoved;
```

Comment:
oh. a stray field :)

Reply:
A result of porting bad design from original powershell. The issue is that we generate only one reg file for all DLLs we process in a build. Thus it does not make sense to run the reg file in next build for each DLL; running it once is sufficient. However, that state has to be persisted between the calls soooo...... 😕